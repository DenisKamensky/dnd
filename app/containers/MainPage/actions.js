/*
 *
 * MainPage actions
 *
 */

import * as types from './constants';

export function defaultAction() {
  return {
    type: types.DEFAULT_ACTION,
  };
}

export function calculateBlockPosition(...args) {
  return {
    type: types.CALCULATE_BLOCK_NEW_POSITION,
    payload: args,
  };
}

export function saveBlockPosition(payload) {
  return {
    type: types.SAVE_NEW_POSITION,
    payload,
  };
}
