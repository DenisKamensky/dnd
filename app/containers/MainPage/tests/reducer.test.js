import { fromJS } from 'immutable';
import mainPageReducer, { blocks } from '../reducer';
import * as actions from '../actions';

describe('mainPageReducer', () => {
  it('returns the initial state', () => {
    expect(mainPageReducer(undefined, {})).toEqual(fromJS({ blocks }));
  });

  it('handels save new position', () => {
    expect(
      mainPageReducer(undefined, actions.saveBlockPosition(blocks)),
    ).toMatchSnapshot();
  });
});
