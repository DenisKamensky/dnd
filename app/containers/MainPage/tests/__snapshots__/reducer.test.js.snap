// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainPageReducer handels save new position 1`] = `
Immutable.Map {
  "blocks": Immutable.List [
    Immutable.Map {
      "id": 1,
      "bgc": "#F8BBD0",
      "top": 0,
      "left": 0,
    },
    Immutable.Map {
      "id": 2,
      "bgc": "#E1BEE7",
      "top": 0,
      "left": 120,
    },
    Immutable.Map {
      "id": 3,
      "bgc": "#42A5F5",
      "top": 0,
      "left": 240,
    },
  ],
}
`;
